### PR TITLE
[Reviewer: Krista] Add in very basic FV tests

### PR DESCRIPTION
**Pull request by eleanor-merry**

### CODE REVIEW

File: src/ut/siptest.cpp [hunk 1]
```diff
@@ -593,42 +594,60 @@ void SipTest::log_pjsip_msg(const char* description, pjsip_msg* msg)
...
+    {
+      hss->set_impu_result("tel:" + user, "call", RegDataXMLUtils::STATE_REGISTERED, "");
+    }
   }
-  AoRPair* aor = sdm->get_aor_data(uri, 0);
-  Binding* binding = aor->get_current()->get_binding(contact);
+
+  Bindings all_bindings;
+  Binding* binding = AoRTestUtils::build_binding(sip_uri, time(NULL));
   binding->_uri = contact;
```
Comment (KristaHyer) [inline src/ut/siptest.cpp]: Can we not just pass this into build_binding?

File: src/ut/sprout_fv_test.cpp [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
+// so far as possible, with the external interfaces being mocked/faked out.
+// So far, these tests are not complete, and aren't particularly well structured
+// either - we should rework the S-CSCF UTs to be much more UT based, pull a
+// bunch of those tests into this suite, and do a proper design/set up proper
+// infrastructure for these tests.
+// For now, these tests run through a couple of mainline calls. They don't
+// really check that much about the calls, just that basically the right flows
+// appear to be happening (as verified by looking at Record-/Route headers).
+// This is sufficient for now; but we shouldn't really add any more tests to
+// this suite without stepping back and doing this properly.
```
Comment (KristaHyer) [inline src/ut/sprout_fv_test.cpp]: Very well explained comment
(although the bit at the end feels slightly like wishful thinking 😛 )

File: src/ut/sprout_fv_test.cpp [hunk 3]
```diff
@@ -0,0 +1,727 @@
...
+                                          _fifc_service,
+                                          ifc_configuration,
+                                          3000, // Session continue timeout - different from default
+                                          6000, // Session terminated timeout - different from default
+                                          _sess_term_comm_tracker,
+                                          _sess_cont_comm_tracker
+                                          );
+    _scscf_sproutlet->init();
+
+    // Create the I-CSCF Sproutlets.
```
Comment (KristaHyer) [inline src/ut/sprout_fv_test.cpp]: sproutletS?

File: src/ut/sprout_fv_test.cpp [hunk 4]
```diff
@@ -0,0 +1,727 @@
...
+using testing::MatchesRegex;
+using testing::HasSubstr;
+using testing::Not;
+using testing::_;
+using testing::NiceMock;
+using testing::HasSubstr;
+using ::testing::Return;
+using ::testing::SaveArg;
+
+// TECH-DEBT-TODO:
```
Comment (KristaHyer) [inline src/ut/sprout_fv_test.cpp]: Can we add the doxygen todo here? It's just `@todo` (and then put a third `/` before each line).

File: src/ut/sprout_fv_test.cpp [hunk 5]
```diff
@@ -0,0 +1,727 @@
...
+  for (int ii = 0; ii < num_hops; ++ii)
+  {
+    inject_msg(curr_response);
+
+    ASSERT_EQ(1, txdata_count());
+    RespMatcher(status_code).matches(current_txdata()->msg);
+
+    // Render the received message to a string so we can re-inject it. 64kB
+    // should be enough space for this.
+    char msg_print_buf[0x10000];
```
Comment (KristaHyer) [inline src/ut/sprout_fv_test.cpp]: It feels unlikely that 64kB translates into 10,000 (bytes?)

File: src/ut/sprout_fv_test.cpp [hunk 6]
```diff
@@ -0,0 +1,727 @@
...
+
+  // ---------- Bono sends a 100 Trying to indicate it has received the request.
+  string fresp_bono = respond_to_txdata(current_txdata(), 100);
+  inject_msg(fresp_bono, &tpBono);
+
+  EXPECT_THAT(get_headers(out, "Record-Route"), testing::MatchesRegex(record_route_regex));
+
+  // Send a 200 OK back down the line to finish the transaction. This is so that
+  // AS communication tracking works correctly. There are a total of 5 hops in
+  // total.
```
Comment (KristaHyer) [inline src/ut/sprout_fv_test.cpp]: It feels like this comment should be combined with the one below it.

File: src/ut/sprout_fv_test.cpp (hunk 2 above)
Comment (eleanor-merry) [reply src/ut/sprout_fv_test.cpp]: Nooo...

**Pull request merged**